runner/copy.go

```diff
@@ -75,28 +75,45 @@ func (m *MaterializedChunkRunner) Run() (types.CompletionWatcher, error) {
 			jobWatcher.EndWatch(fmt.Errorf("failed to create iterator: %w", err))
 			return
 		}
-		i := 0
+		// Buffer the records channel to ensure readers won't block the writer loop below
+		ch := make(chan provider.ResourceRecord, 1_000_000)
+		var wg sync.WaitGroup
+		waitGroupSize := 1000
+		wg.Add(waitGroupSize)
+		// Create a "pool" of 100 goroutines to write to the table to better utilize CPU resources
+		// and speed up the write operations to the inference table.
+		for idx := 0; idx < waitGroupSize; idx++ {
+			go func() {
+				defer wg.Done()
+				for record := range ch {
+					err := m.Table.Set(record.Entity, record.Value)
+					if err != nil {
+						close(ch)
```

Comment:
There's a bug in here (that chatgpt helped me find) (and I brought it up to simba too, he said he'll help you with it)

By closing the channel here; if there was an error, it closes all the other channels. Which is fine. BUT, now that the channels are closed, if you try to write to the channel below, you'd get a shitty panic. There's a trick about using an error channel with buffer size of 1 to capture those which allows you to more gracefully handle the error.

Reply:
Addressed and tested.